apps/elixir_ls_utils/test/support/mix_test.case.ex

```diff
@@ -2,6 +2,14 @@ defmodule ElixirLS.Utils.MixTest.Case do
   # This module is based heavily on MixTest.Case in Elixir's tests
   use ExUnit.CaseTemplate
 
+  defp mix_state do
```

Comment:
Rather than this function which is returning an opaque module I think it would be better to define a `private_clear_cache/0` that is instead of the previous `Mix.ProjectStack.clear_cache()`. That way the blast-zone of the private api usage is easier to track. Also it would be good to note the private api usage with a comment, although I'm working on a separate PR for that.